### PR TITLE
Add copy pkg

### PR DESCRIPTION
**Pull request by tonistiigi**
Golang replacement for https://github.com/tonistiigi/copy to support `chown`. I'll add similar one for archive as well.

### CODE REVIEW

File: copy/copy.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+		}
+		if err := os.Symlink(link, target); err != nil {
+			return errors.Wrapf(err, "failed to create symlink: %s", target)
+		}
+	case (fi.Mode() & os.ModeDevice) == os.ModeDevice:
+		if err := copyDevice(target, fi); err != nil {
+			return errors.Wrapf(err, "failed to create device")
+		}
+	default:
+		// TODO: Support pipes and sockets
```
Comment (AkihiroSuda) [inline copy/copy.go]: Is it technically feasible?

File: copy/copy_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+	t2, err := ioutil.TempDir("", "test")
+	require.NoError(t, err)
+	defer os.RemoveAll(t2)
+
+	err = Copy(context.TODO(), filepath.Join(t1, "foo*"), t2)
+	require.Error(t, err)
+
+	err = Copy(context.TODO(), filepath.Join(t1, "foo*"), t2, AllowWildcards)
+	require.NoError(t, err, fmt.Sprintf("%+v", err))
```
Comment (AkihiroSuda) [inline copy/copy_test.go]: nit: extra Sprintf

File: copy/copy.go (hunk 1 above)
Comment (tonistiigi) [reply copy/copy.go]: This is from containerd copyDir source. Pipes should definitely be possible, for sockets could just create empty files.

**Comment by AkihiroSuda** [comment]
@tonistiigi is this mergeable?

**Comment by tonistiigi** [comment]
@AkihiroSuda Yes, I think so. Rebased and fixed `fstest` import.

**Pull request merged**